lib/internal/Magento/Framework/Controller/Result/Json.php

```diff
@@ -29,24 +29,63 @@ class Json extends AbstractResult
     protected $json;
 
     /**
-     * @param \Magento\Framework\Translate\InlineInterface $translateInline
+     * @var \Magento\Framework\Serialize\Serializer\Json
      */
-    public function __construct(InlineInterface $translateInline)
-    {
+    private $serializer;
+
+    /**
+     * @param InlineInterface $translateInline
+     * @param \Magento\Framework\Serialize\Serializer\Json|null $serializer
+     * @throws \RuntimeException
+     */
+    public function __construct(
+        InlineInterface $translateInline,
+        \Magento\Framework\Serialize\Serializer\Json $serializer = null
```

Comment:
Thanks, I expected something like this. Not sure why such interface was introduced at all if we keep creating json-only code.

Needs to be mentioned somewhere to avoid confusion like https://github.com/magento/magento2/pull/10335#discussion_r129590760